refactor: Rework Core Metadata docs to match updated 3.1 structure (#1257)
edgexfoundry/edgex-docs: docs_src/microservices/core/metadata/details/DeviceSystemEvents.md

---
title: Core Metadata - Device System Events
---

# Core Metadata - Device System Events

> Device System Events are events triggered by the add, update or delete of devices. A System Event DTO is published to the EdgeX MessageBus each time a new Device is added, an existing Device is updated or when an existing Device is deleted.

Reviewer: ```suggestion
System Events are events triggered by the add, update or delete of device metadata objects (Device, DeviceProfile, etc.). A System Event DTO is published to the EdgeX MessageBus each time a new object is added, an existing object is updated or when an existing objects is deleted.
```

> ### System Event DTO

Reviewer: ```suggestion
## System Event DTO
```

| --------- | --------------------------------------------- | ------------------------------------------------------------ |
| Type      | Type of System Event                          | `device`, `deviceservice`,  `deviceprofile`, or `provisionwatcher`                                       |
| Action    | System Event action                           | `add`, `update`, or `delete` in this case                    |
| Source    | Source of the System Event                    | `core-metadata` in this case                                 |
| Owner     | Owner of the data in the System Event         | In this case it is the name of the device service that owns the device or `core-metadata` |
| Tags      | Key value map of additional data              | empty in this case                                           |
| Details   | The data object that trigger the System Event | the added, updated, or deleted Device/Device Profile/Device Service/Provision Watcher in this case           |
| Timestamp | Date and time of the System Event             | timestamp                                                    |

> ### Publish Topic

Reviewer: ```suggestion
## Publish Topic
```
